src/payment/strategies/affirm/affirm-payment-strategy.ts

```diff
@@ -107,36 +109,42 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
             throw new MissingDataError(MissingDataErrorType.MissingCheckout);
         }
 
-        if (!consignments) {
-            throw new MissingDataError(MissingDataErrorType.MissingConsignments);
-        }
-
-        const consignment = consignments[0];
-
-        if (!consignment || !consignment.selectedShippingOption) {
-            throw new MissingDataError(MissingDataErrorType.MissingConsignments);
-        }
+        const amountTransformer = new AmountTransformer(order.currency.decimalPlaces);
+        const billingAddress = this._getBillingAddress();
 
         return {
             merchant: {
                 user_confirmation_url: config.links.checkoutLink,
                 user_cancel_url: config.links.checkoutLink,
                 user_confirmation_url_action: 'POST',
             },
-            shipping: this._getShippingAddress(),
-            billing: this._getBillingAddress(),
-            items: this._getItems(),
-            discounts: this._getDiscounts(),
+            shipping: this._getShippingAddress() || billingAddress,
+            billing: billingAddress,
+            items: this._getItems(amountTransformer, order),
             metadata: {
-                shipping_type: consignment.selectedShippingOption.type,
+                shipping_type: this._getShippingType(consignments),
                 mode: 'modal',
+                platform_type: 'BigCommerce',
+                platform_version: '',
+                platform_affirm: '',
             },
+            discounts: this._getDiscounts(amountTransformer, order),
             order_id: order.orderId ? order.orderId.toString() : '',
-            shipping_amount: order.shippingCostTotal * 100,
-            tax_amount: order.taxTotal * 100,
-            total: order.orderAmount * 100,
+            shipping_amount: amountTransformer.toInteger(order.shippingCostTotal),
+            tax_amount: amountTransformer.toInteger(order.taxTotal),
+            total: amountTransformer.toInteger(order.orderAmount),
         };
+    }
 
+    private _getShippingType(consignments: Consignment[] | undefined): string {
```

Comment:
`consignments?: Consignment[]` instead of `undefined` if you want an optional parameter.

Reply:
@michxff this change doesn't appear to have been addressed.